### PR TITLE
CI (python matrix): Update to 3.10

### PR DESCRIPTION
**Pull request by tony**
Resolves: https://github.com/python-poetry/poetry/issues/5087

- [x] Added **tests** for changed code. N/A
- [x] Updated **documentation** for changed code. N/A

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (danieleades) [inline .github/workflows/tests.yml:25]: why is this marked `experimental` in the 'tests' workflow, but not the 'integration' workflow?

**Review by danieleades — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml
Comment (tony) [reply .github/workflows/tests.yml:25]: Unlike the integration test, this points to master poetry git repo. Here's the block in full from `tests.yml`:

```yaml
          - os: Ubuntu
            python-version: "3.10"
            experimental: true
            bootstrap-args: "--git https://github.com/python-poetry/poetry.git"
```

So the question may be: Why doesn't tox also check against git master? I propose that could be better to tackle in a separate issue, but we can also do it here if there's an agreement we could come to.

`workflow/integration.yml`:

[Integration test action](https://github.com/python-poetry/poetry-core/blob/25f6ded/.github/workflows/integration.yml#L31-L33) ([tox.ini](https://github.com/python-poetry/poetry-core/blob/25f6dedcd142148daf3e2f4f495d30055715ddd7/tox.ini)) run [tests/integration/](https://github.com/python-poetry/poetry-core/tree/c6dd2487947a450b030f9649c931964ac4e4a83a/tests/integration).

The reason it had `--experimental` before may be because 3.10 support itself wasn't stable.

Comment (danieleades) [reply .github/workflows/tests.yml:25]: that makes sense. might be good to raise a separate ticket for discussion about the potential inconsistency

**Review by danieleades — DISMISSED** [review]
looks ok to me. I don't have perms to trigger the CI i'm afraid

**Comment by tony** [comment]
@danieleades 

> @tony tony dismissed danieleades’s stale review via e115389 now

Same, just a rebase

**Comment by finswimmer** [comment]
@tony Could you please rebase :pray:

File: .github/workflows/integration.yml
Comment (branchvincent) [inline .github/workflows/integration.yml:13]: just needs to be quoted (else it's interpreted as the float `3.1`):
```suggestion
        python-version: [3.7, 3.8, 3.9, "3.10"]
```

File: .github/workflows/tests.yml
Comment (branchvincent) [inline .github/workflows/tests.yml:16]:
```suggestion
        python-version: [3.7, 3.8, 3.9, "3.10"]
```

File: .github/workflows/integration.yml
Comment (tony) [reply .github/workflows/integration.yml]: Good catch!

File: .github/workflows/tests.yml
Comment (tony) [reply .github/workflows/tests.yml]: [763240f](https://github.com/python-poetry/poetry-core/pull/265/commits/763240f96c80432f012c705234b86ce2a9f7061f)

File: .github/workflows/integration.yml
Comment (branchvincent) [inline .github/workflows/integration.yml:13]: seems everything is passing but let's keep `experimental` as it's referenced above:
```suggestion
        python-version: [3.7, 3.8, 3.9, "3.10"]
        experimental: [false]
```

Comment (tony) [reply .github/workflows/integration.yml:13]: I think we should remove `continue-on-error: ${{ matrix.experimental }}` or keep it as is.

The reason why is python 3.10 is stable now and considered officially supported in poetry.

I think experimental would be for either or both of these conditions:
- Alpha python release, e.g. 3.11 alpha
- python release unsupported by poetry, e.g. if 3.10 wasn't considered ready for poetry despite python release 3.10.0 officially
- Bootstrapping poetry trunk, e.g. `bootstrap-args: "--git https://github.com/python-poetry/poetry.git"` (but `tox.ini` doesn't support that and it may be best to defer that to a separate PR)

Comment (branchvincent) [reply .github/workflows/integration.yml:13]: agreed, let's remove `continue-on-error: ${{ matrix.experimental }}` for now. 

We'll prob add it back to test 3.11 alpha/beta in the future like you said, but we can handle that then

Comment (tony) [reply .github/workflows/integration.yml:12]: [c3bc556](https://github.com/python-poetry/poetry-core/pull/265/commits/c3bc556a0e7b683553f3a824a13077fcb644322e)

**Review by branchvincent — APPROVED** [review]
@tony thanks again! LGTM but like last time, we'll have to wait for another maintainer to merge

**Comment by tony** [comment]
@branchvincent Thank you for the review!

I will sit tight and rebase as needed to make sure the PR stays fresh 👍

**Review by finswimmer — APPROVED** [review]
Thanks a lot :+1:

**Comment by tony** [comment]
@finswimmer always.

**Pull request merged**
